### PR TITLE
[ARTS-758] - Validation of configuration files (Part 2)

### PR DESCRIPTION
**Pull request by eladiw**
## Description
Adding schema files for all business micro services + updating trial def schema + adding validation steps

NOTE: This PR should be merged TOGETHER with https://github.com/NDPH-ARTS/mts-services/pull/207

Result:
image

### Changes
- [ARTS-758] - Validation of configuration files

### Checklist:

- [x] Branch name follows convention (feature/arts-###-short-name OR fix/arts-###-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-758]: https://ndph-arts.atlassian.net/browse/ARTS-758

### CODE REVIEW

**Review by anatbal — DISMISSED** [review]

File: .github/workflows/terraform-plan-apply.yml [hunk 1]
```diff
@@ -15,13 +15,51 @@ jobs:
...
+          set -o nounset
+
           sudo npm install -g ajv-cli
-          find . -name definition.yml -exec ajv validate -s $INPUT_SCHEMA -d {} \;
+
+          export FILES=$(find . -name 'definition.yml')
+          for FILE in $(echo "$FILES" | sed "s/,/ /g")
+          do
+               ajv validate -s $INPUT_SCHEMA -d "$FILE"
+          done
```
Comment (tamirkamara) [inline .github/workflows/terraform-plan-apply.yml:33]: Why do you need to run a for loop? Isn't the "before" way with the find and "exec" enough?

File: .github/workflows/terraform-plan-apply.yml [hunk 2]
```diff
@@ -15,13 +15,51 @@ jobs:
...
+        run: |
+          set -o errexit
+          set -o pipefail
+          set -o nounset
+
+          export FILES=$(find . -name 'init-service-dev.yml')
+          for FILE in $(echo "$FILES" | sed "s/,/ /g")
+          do
+               ajv validate -s $INIT_SCHEMA -d "$FILE"
+          done
```
Comment (tamirkamara) [inline .github/workflows/terraform-plan-apply.yml]: This only looks for "dev" config files but in the future we could see other profiles too. Probably should look for and init-service*.yml, right?

Same comment on the other config files below and also the use of find/exec

File: trials/ci-trial/services/site-service-dev.yml [hunk 3]
```diff
@@ -4,7 +4,6 @@ server:
 fhir:
   resultCount: 50
 
----
 # this is the default 'development' trial site config - CCO/REGION/COUNTRY/LCC
 address:
```
Comment (tamirkamara) [inline trials/ci-trial/services/site-service-dev.yml]: Shouldn't this be also under mts?

**Review by tamirkamara — CHANGES_REQUESTED** [review]

File: .github/workflows/terraform-plan-apply.yml (hunk 1 above)
Comment (eladiw) [reply .github/workflows/terraform-plan-apply.yml:33]: didn't work due to the exit code not being picked up

**Review by tamirkamara — APPROVED** [review]

**Review by anatbal — APPROVED** [review]

**Pull request merged**